### PR TITLE
feat: create store locator template

### PR DESCRIPTION
**Pull request by sohnj877**
Screenshot 2023-11-29 at 4 43 25 PM

### CODE REVIEW

**Review by bashunaimiroy — CHANGES_REQUESTED** [review]

**Review by bashunaimiroy — APPROVED** [review]
approved, just left one question re. event names

**Review by bashunaimiroy — APPROVED** [review]

File: site/global/sections/footer.json [hunk 1]
```diff
@@ -48,7 +48,7 @@
                     {
                         "label": "Locations And Hours",
                         "type": "external",
```
Comment (kchung) [inline site/global/sections/footer.json:50]: Can we start using the `Link` resource shape + the `link` function? 

There was [recently some documentation added for this](https://square.github.io/custom-sites-docs/template-syntax/functions/link) and we should document the shape of the `Link` resource in its own section (it's the same as the function doc)

In this case, it'd be:

```
{
  "type": "storeLocator"
}
```

and ideally consumed like

```twig
<a href="{{ link(payload) }}">Store locator</a>
```

File: site/global/sections/footer.json [hunk 2]
```diff
@@ -47,8 +47,8 @@
                 "children": [
                     {
                         "label": "Locations And Hours",
-                        "type": "external",
-                        "href": "/"
+                        "type": "storeLocator",
+                        "href": "/s/store-locator"
```
Comment (kchung) [inline site/global/sections/footer.json]: this is considered a "system page" link, so you do not need to specify an `href` value

```suggestion
                        "type": "storeLocator"
```

https://square.github.io/custom-sites-docs/template-syntax/functions/link#system-links

File: site/pages/store-locator.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+    "props": {
+        "locations": {
+            "filters": []
+        }
+    },
+    "template": "templates\/store\/locations",
+    "route": "s\/store-locator"
```
Comment (kchung) [inline site/pages/store-locator.json]: do these need to be escaped? does it work if escaped?

Comment (sohnj877) [reply site/pages/store-locator.json]: I think these were added automatically when I downloaded the files 🤔  will remove

**Review by kchung — APPROVED** [review]

**Pull request merged**